tests/relational/test_relational_data_with_json.py

```diff
@@ -192,47 +220,74 @@ def test_invented_json_column_names(documents, bball):
     ]
 
 
-def test_set_some_primary_key_to_none(documents, bball):
+@patch("gretel_trainer.relational.json.make_suffix")
+def test_set_some_primary_key_to_none(mock_make_suffix, documents):
+    # We need to reset the side_effect iterator later in the test, so we manually mock it
+    mock_make_suffix.side_effect = [f"sfx{i}" for i in range(1, 50)]
```

Comment:
I think if you update the conftest implementation to yield the mock object (instead of just yielding nothing), you can avoid having to patch each of these tests and (re)define the initial implementation at the top of each of them. You'll still need to reset the side effect at the appropriate time (right before calling `set_primary_key`), of course. Something like this (I'm short-handing the fixture a bit, this example doesn't have your new changes)

```python
# conftest
def static_suffix():
    with patch("gretel_trainer.relational.json.make_suffix") as make_suffix:
        make_suffix.side_effect = itertools.count
        yield make_suffix  # instead of just `yield` here


# these tests
def test_set_some_primary_key_to_none(static_suffix, documents):
    # setup stuff, should take advantage of the autouse fixture from conftest

    # but now we have a handle to it and can reset it here
    static_suffix.side_effect = itertools.count
    documents.set_primary_key(...)

Reply:
Makes sense, cooking up a revision.